week1-project/ecdsa-node/server/index.js

```diff
@@ -1,31 +1,90 @@
-const express = require("express");
+/* eslint-disable no-console */
+const express = require('express');
+
 const app = express();
-const cors = require("cors");
+const cors = require('cors');
+const secp = require('ethereum-cryptography/secp256k1');
+const { toHex } = require('ethereum-cryptography/utils');
+
 const port = 3042;
+const { utf8ToBytes } = require('ethereum-cryptography/utils');
 
 app.use(cors());
 app.use(express.json());
 
-const balances = {
-  "0x1": 100,
-  "0x2": 50,
-  "0x3": 75,
-};
+const actors = ['zahra', 'kate', 'constance', 'martina'];
+const actorKeyPairs = {};
+const balances = {};
+
+function generateKeyPairs() {
+  actors.forEach((item) => {
+    const privateKey = toHex(secp.utils.randomPrivateKey());
+    actorKeyPairs[`${item}`] = {
+      private_key: privateKey,
+      public_key: toHex(secp.getPublicKey(privateKey)),
+    };
+  });
+  return actorKeyPairs;
+}
+generateKeyPairs();
```

Comment:
Given that you invoke this right after defining it, it might be a good fit for an [`iife`](https://developer.mozilla.org/en-US/docs/Glossary/IIFE).

Reply:
Ah, looking more closely at this... you don't do anything with the return value and you're relying on mutating an external variable. This would be better restructured as:
```suggestion
const actorKeyPairs = generateKeyPairs();
```
Then `generateKeyPairs` itself could use a `reduce` over actors with an initial accumulator of `{}`. Eh, an example would probably help:

```js
const generateKeyPairs = () => {
  return actors.reduce((acc, actor) => acc[actor] = {
    private_key: privateKey,
    public_key: toHex(secp.getPublicKey(privateKey))
  }, {});
}
const actorKeyPairs = generateKeyPairs();
```
👆 personal preference, but I like the more functional approach.

Reply:
Ok. I get what you are saying here but I was struggling to figure out how to get the privateKey (which needs to be generated for each actor) to be included in the accumulated actor hash and still pass it to the getPublicKey. I am sure this is doable but I wasn't able to figure it out.

Reply:
I'm going to move forward to my next project but maybe you and I can circle back to this in a pairing session.

Reply:
For sure, and I'm happy to pair. Does the `reduce` example make sense though? I think it should be a drop-in replacement.

Reply:
It makes sense but because of the privateKey issue I was describing above it didn't work as a drop in replacement. We can pair on it though.